tests/testthat/test-factor-level-coercion.R

```diff
@@ -0,0 +1,926 @@
+
+# Add noise
+df$hotDogScore <- df$hotDogScore + rnorm(n, mean = 0, sd = 1.25)
+
+# Regression data set
+dfR <- df
+dfR$isHotDog <- NULL
+dfR$vendorID <- NULL
+
+# Set parameters for regression models
+pR <- SupervisedModelDevelopmentParams$new()
+pR$df <- dfR
+pR$type <- "regression"
+pR$impute <- TRUE
+pR$grainCol <- "id"
+pR$predictedCol <- "hotDogScore"
+pR$debug <- FALSE
+pR$cores <- 1
+
+# Run regression models
+RandomForestR <- RandomForestDevelopment$new(pR)
+capture.output(RandomForestR$run())
+LassoR <- LassoDevelopment$new(pR)
+capture.output(LassoR$run())
+
+# Classification data set
+dfC <- df
+# assign response variable
+dfC["isHotDog"] <- ifelse(dfC["hotDogScore"] > 0, "Y", "N")
+dfC$isHotDog <- as.character(dfC$isHotDog)
+dfC$hotDogScore <- NULL
+dfC$vendorID <- NULL
+
+# Set parameters for classification models
+pC <- SupervisedModelDevelopmentParams$new()
+pC$df <- dfC
+pC$type <- "classification"
+pC$impute <- TRUE
+pC$grainCol <- "id"
+pC$predictedCol <- "isHotDog"
+pC$debug <- FALSE
+pC$cores <- 1
+
+# Run classification models
+RandomForestC <- RandomForestDevelopment$new(pC)
+capture.output(RandomForestC$run())
+LassoC <- LassoDevelopment$new(pC)
+capture.output(LassoC$run())
+
+# LMM data set
+dfLMM <- df
+# Shift values based on vendorID: higher vendorID -> greater chance of hotdog
+dfLMM$hotDogScore <- dfLMM$hotDogScore + rnorm(n, 
+                                               mean = (dfLMM$vendorID - 5)/2, 
+                                               sd = 0.25)
+# assign response varaible
+dfLMM["isHotDog"] <- ifelse(dfLMM["hotDogScore"] > 0, "Y", "N")
+dfLMM$isHotDog <- as.character(dfLMM$isHotDog)
+dfLMM$hotDogScore <- NULL
+
+# Set parameters for LMM
+pL <- SupervisedModelDevelopmentParams$new()
+pL$df <- dfLMM
+pL$type <- "classification"
+pL$impute <- TRUE
+pL$grainCol <- "id"
+pL$personCol <- "vendorID"
+pL$predictedCol <- "isHotDog"
+pL$debug <- FALSE
+pL$cores <- 1
+
+# Run LMM models
+LMM <- LinearMixedModelDevelopment$new(pL)
+capture.output(LMM$run())
+
+# reset seed
+set.seed(NULL)
+
+#### BEGIN TESTS ####
+
+#### CLASSIFICATION TESTS ####
+
+test_that("Single row predictions work for RF classification", {
+  # Single row prediction data set
+  dfDeploy1 <- data.frame(id = 9001,
+                          length = 6,
+                          diameter = 3,
+                          heat = "Cold",
+                          condiment = "Syrup")
+  
+  p1 <- SupervisedModelDeploymentParams$new()
+  p1$type <- "classification"
+  p1$df <- dfDeploy1
+  p1$grainCol <- "id"
+  p1$predictedCol <- "isHotDog"
+  p1$impute <- TRUE
+  p1$debug <- F
+  p1$cores <- 1
+  
+  # Deploy rf on single row
+  capture.output(rfD1 <- RandomForestDeployment$new(p1))
+  rfD1$deploy()
+  
+  # Get prediction
+  rfOutDf1 <- rfD1$getOutDf()
+  
+  # Check that dimensions of prediction dataframe are correct
+  expect_equal(dim(rfOutDf1)[1], 1)
+  expect_equal(dim(rfOutDf1)[2], 8)
```

Comment:
I changed this to test 1) whether the prediction output dataframe has the same grainCol id as the deploy dataframe and 2) whether the predicted value/probability is numeric.  It still doesn't tell you if the prediction is reasonable, but that is done to a certain extent in the next test (check whether predictions are independent) by checking that the predictions are consistent with what would have occurred using the in-test flag.

Checking the data type of the prediction also revealed a problem with developing a random forest regressor and a random forest classifier at the beginning of the same test file (one overwrote the other).